ios/RNTypeValidation.m

```diff
@@ -0,0 +1,211 @@
+        BOOL isLineTwoValid = [address[@"line2"] isKindOfClass:[NSString class]] || address[@"line2"] == nil;
+        BOOL isLineThreeValid = [address[@"line3"] isKindOfClass:[NSString class]] || address[@"line3"] == nil;
+        BOOL isPostCodeValid = [address[@"postCode"] isKindOfClass:[NSString class]] || address[@"postCode"] == nil;
+        BOOL isTownValid = [address[@"town"] isKindOfClass:[NSString class]] || address[@"town"] == nil;
+        BOOL isCountryCodeValid = [address[@"countryCode"] isKindOfClass:[NSString class]] || address[@"countryCode"] == nil;
+        return isLineOneValid && isLineTwoValid && isLineThreeValid && isPostCodeValid && isTownValid && isCountryCodeValid;
+    } else {
+        return NO;
+    }
+}
+
++ (BOOL)isUIConfigurationsValid:(NSMutableDictionary *)uiConfigurations {
+    if ([uiConfigurations isKindOfClass: [NSMutableDictionary class]] || uiConfigurations == nil) {
+        BOOL isAVSEnabledValid = [uiConfigurations[@"isAVSEnabled"] isKindOfClass: [NSNumber class]];
+        BOOL shouldDisplayAmountValid = [uiConfigurations[@"shouldDisplayAmount"] isKindOfClass: [NSNumber class]];
+        BOOL isThemeValid = [uiConfigurations[@"theme"] isKindOfClass: [NSMutableDictionary class]] ||uiConfigurations[@"theme"] == nil ;
+        return isAVSEnabledValid && shouldDisplayAmountValid && isThemeValid;
+    }  else {
+        return NO;
+    }
+}
+
++ (BOOL)isPaymentSummaryItemsValid:(NSMutableArray *)summaryItems {
+    if([summaryItems isKindOfClass:[NSMutableArray class]]){
+        for (NSMutableDictionary *item in summaryItems) {
+            BOOL isLabelValid = [item[@"label"] isKindOfClass: [NSString class]];
+            BOOL isAmountValid = [item[@"amount"] isKindOfClass: [NSString class]];
+            BOOL isTypeValid = [item[@"type"] isKindOfClass: [NSNumber class]] || item[@"type"] == nil;
+            BOOL isItemValid = isLabelValid && isAmountValid && isTypeValid;
+            if(!isItemValid) {
+                return NO;
+                break;
+            }
+        }
+        return YES;
+    } else {
+        return NO;
+    }
+}
+
++ (BOOL)isShippingMethodsValid:(NSMutableArray *)shippingMethods {
+    if([shippingMethods isKindOfClass:[NSMutableArray class]]){
+        for (NSMutableDictionary *item in shippingMethods) {
+            BOOL isIdentifierValid = [item[@"identifier"] isKindOfClass: [NSString class]];
+            BOOL isDetailValid = [item[@"detail"] isKindOfClass: [NSString class]];
+            BOOL isLabelValid = [item[@"label"] isKindOfClass: [NSString class]];
+            BOOL isAmountValid = [item[@"amount"] isKindOfClass: [NSString class]];
+            BOOL isTypeValid = [item[@"type"] isKindOfClass: [NSNumber class]];
+            BOOL isItemValid = isIdentifierValid && isDetailValid && isLabelValid && isAmountValid && isTypeValid;
+            if(!isItemValid) {
+                return NO;
+                break;
```

Comment:
There's no need for a `break` statement since we have `return NO` above it. It will never execute